tests/pytests/test_vecsim.py

```diff
@@ -2394,3 +2390,35 @@ def test_switch_write_mode_multiple_indexes(env):
     if bg_indexing == 0:
         prefix = "::warning title=Bad scenario in test_vecsim:test_switch_write_mode_multiple_indexes::" if GHA else ''
         print(f"{prefix}All vectors were done reindex before switching back to in-place mode")
+
+
+def test_vector_index_ptr_valid(env):
+    conn = getConnectionByEnv(env)
+    # Scenerio1: Vecsim Index scheme with numeric (or non-vector type) and vector type with invalid parameter 
+    #            Insert partial doc - only numeric
+    #            Update Doc
+    
+    # HNSW parameters the causes an execution throw (M > UINT16_MAX)
+    UINT16_MAX = 2**16
+    M = UINT16_MAX + 1
+    dim = 4
+
+    env.expect('FT.CREATE', 'idx','SCHEMA', 'n', 'NUMERIC',
+                    'v', 'VECTOR', 'HNSW', '8', 'TYPE', 'FLOAT16', 'DIM', dim, 'DISTANCE_METRIC', 'L2', 'M', M).ok()
+
+    res = conn.execute_command('HSET', 'doc', 'n', 0)
+    env.assertEqual(res, 1)
+    # Before bug fix, the following command would cause a server crash due to the null pointer accsess
+    res = conn.execute_command('HSET', 'doc', 'n', 1)
+    env.assertEqual(res, 0)
+    
+    # Sanity check - insert a vector, expect indexing faliure
+    res = conn.execute_command('HSET', 'doc1', 'v', create_np_array_typed([0]*dim,'FLOAT16').tobytes())
+    env.assertEqual(res, 1)
+
+    index_errors_dict = index_errors(env, 'idx')
+    env.assertEqual(index_errors_dict['last indexing error'], "Could not open vector for indexing")
+
+    # Check FlushAll OK - before bug fix, the following command would cause a server crash due to the null pointer accsess
+    env.expect('FLUSHALL').noError()
```

Comment:
BTW, before the fix, we got OK from flushall, and then crashed in the background thread no?

Reply:
Yes that's right, that means that the test will pass even if the bug is still there.
maybe I should a ping after flushall and expect a pong?

Reply:
Yes that's a good idea.
The test would have probably failed anyway since when the test ends there is another call to `FLUSHALL` by RLTest, but it is still better to rely on our check rather than the infra :)